docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,398 @@
+        # y now points to z_1.select(0, 0)
+        return y
+
+The corresponding autograd computations can simply be written using the implicit variables: :math:`v^T J_f = v^T J_{op_2\_} J_{op_1} = v^T \dfrac{\partial{z_1}}{\partial{z_0}} \dfrac{\partial{z_0}}{\partial{x}} = v^T \dfrac{\partial{z_1}}{\partial{x}}`.
+And for g, it is harder to write as only part of :code:`op_2_()` is used when computing :code:`y`, but it can be written as :math:`v^T J_g = v^T \dfrac{\partial{y_1}}{\partial{y_0}} \dfrac{\partial{y_0}}{\partial{z_0}} \dfrac{\partial{z_0}}{\partial{x}}`.
+
+Computing not gradients
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In some cases, the user might need to compute something that is not the "true gradient" of its function.
+We identify few use cases here:
+
+- Ignore the gradients for some part of the computation as they could render the computed gradients unstable, e.g. gradients of square root near :math:`0`.
+- Ignore the gradients in a set of computations where we know they will be :math:`0` and so don't need to be computed, e.g. a set of operations that compute values that are the used as integers to index a tensor.
+- Update the value of a tensor that requires gradients without this update being recorded: initialization of the weights of a neural network.
+- Work with tensors that may or may not require gradients to perform operations that do not need gradients computed: optimizers for neural networks or method that compute the 0-1 accuracy for classification.
+
+To be able to perform all these operations, we provide three main constructs.
+
+Detaching
+"""""""""
+
+The first construct is the method :code:`y = x.detach()` that can be used on any tensor.
+This is a special function for which :math:`J_{detach} = \bm{0}`.
+This means that however :code:`y` is used, the contribution via :code:`y` to the gradient of :code:`x` is always going to be :math:`0`.
+
+.. code::
+
+    def f(x):
+        z = op_1(x)
+
+        y = z.detach()
+        return y
+
+    # x is a Tensor of the appropriate size
+    x.requires_grad_()
+
+    # v is a Tensor of the appropriate size
+    f(x).backward(v)
+
+The backward pass will compute the following for a given vector :math:`v`: :math:`v^T J_f = v^T J_{detach} J_{op_1} = v^T \bm{0} \dfrac{\partial{z}}{\partial{x}} = \bm{0}`.
+
+Note that the new Tensor shares the same data as the original one (it is a view).
+This means that inplace operations on :code:`y` will change values in :code:`x` .
+
+Disable gradient computation
+""""""""""""""""""""""""""""
+
+The second construct is the function decorator :code:`torch.no_grad()` (it can also be used as a context manager).
+When it decorates a function :code:`outputs = f(inputs)`, it is defined as enforcing :math:`\forall in \in inputs, \forall out \in outputs, \dfrac{\partial{out}}{\partial{in}} = 0`.
+Note that for the context manager version, outputs are all the variables that are assigned (on the left side of an :code:`=` in the block).
```

Comment:
But traditionally, `no_grad` isn't used to selectively turn off gradients inside an autograd computation; it's just used to let you run a model which might have some `requires_grad` inputs in inference mode. The framing in this section doesn't make that very clear.